### PR TITLE
docs: added Merkle tree use recommendations

### PR DESCRIPTION
**Pull request by prodasl**
## Description
Added FAQ to guide the developers about when and why use a specific type of Merkle tree for their projects

The FAQ has a short text answer plus a comparison table with more detailed data Refs: #311

## Related Issue(s)
docs: add section to help choose merkle tree [#311](https://github.com/privacy-scaling-explorations/zk-kit/issues/311)

## Checklist
It would be helpful to write a comparison of of all them, answering questions like
-   [x] "when not to use"/"when to use"
-   [x]  "used by?": protocol xyz, app xyz...
-   [ ] include benchmarks: maybe one merkle tree is faster at generating a proof, but slower to update?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -499,3 +499,14 @@ grep -r -l "smt" . | xargs sed -i 's/smt/my-package/'
...
+|-----------------|---------------------------|----------------------------------------------|-----------------------------------------------|----------------------------------------------|------------------------------------------------|-----------------------|
+|**Incremental**  | @zk-kit/imt               | Fastest for incremental updates              | Applications requiring frequent updates       | Scenarios with very large datasets           | Semaphore, Tornado Cash                        | Fast                  |
+|**Lean incremental**| @zk-kit/lean-imt          | Optimized for lightweight environments       | Mobile and browser-based ZK applications      | High-frequency update requirements           | zkSync, Loopring                                | Moderate              |
+|**Sparse**           | @zk-kit/smt               | Handles very large sets efficiently          | Applications with large static datasets       | Frequent updates due to higher computational cost | Aztec, Zcash                   | Slow (due to large set handling) |
```
Comment (sripwoud) [inline README.md:512]: what about adding links to the homepage of the projects listed?

File: README.md [hunk 2]
```diff
@@ -499,3 +499,14 @@ grep -r -l "smt" . | xargs sed -i 's/smt/my-package/'
 #### How can I create benchmarks for my library?
 
 You can see some examples in the `benchmarks` folder. All you have to do is create a file that exports a function to run your benchmark in that folder, and add that function to the `index.ts` file. The `yarn benchmarks` command can be run with no parameters (it will run all the benchmarks), or you can specify the name of your benchmark file to run just that. When you run the command it will create a `benchmarks/results` folder with your results.
+
+#### I need to use a Merkle Tree to prove the inclusion or exclusion of data elements within a set. Which type of Merkle Tree should I use?
+**Incremental:** Ideal for applications where you frequently add new elements and need to update the tree efficiently.
+***Lean Incremental:** A more memory-efficient version of the incremental Merkle tree, suitable for use cases with memory constraints.
+**Sparse:** Best suited for scenarios where you need to manage a large, mostly empty set of possible elements.
```
Comment (sripwoud) [inline README.md:506]: I think we are missing new lines here, The formatting is a bit off
image
```suggestion
**Incremental:** Ideal for applications where you frequently add new elements and need to update the tree efficiently.  
***Lean Incremental:** A more memory-efficient version of the incremental Merkle tree, suitable for use cases with memory constraints.  
**Sparse:** Best suited for scenarios where you need to manage a large, mostly empty set of possible elements.
```

**Review by sripwoud — CHANGES_REQUESTED** [review]
I love the table!
Just a small request for change regarding the formatting.  
And make sure to check the formatting to ensure the CI check passes (see [code quality and formatting](https://github.com/privacy-scaling-explorations/zk-kit?tab=readme-ov-file#code-quality-and-formatting))

What about benchmarks? Would you like tackling that in a later PR too?
Collecting data like this would be amazing (we could easily build nice charts from it):

|Tree Type|size (MB and/or leaves and/or tree depth )|operation (insert, update ...)|time (ms)|
|---|---|---|---|

File: README.md (hunk 1 above)
Comment (prodasl) [reply README.md:512]: Got it, working on it

File: README.md (hunk 2 above)
Comment (prodasl) [reply README.md:506]: Working on it

**Comment by prodasl** [comment]
Sure, I can send it in another PR if you think it is a good idea

**Comment by prodasl** [comment]
I have already made the changes of formatting and the home pages URL for the projects, please help me with the review

**Review by sripwoud — APPROVED** [review]
ACK
@cedoor this looks good to merge to me.

**Pull request merged**